objectRecognitionSource/src/test/java/io/vantiq/extsrc/objectRecognition/neuralNet/TestYoloProcessor.java

```diff
@@ -131,6 +139,129 @@ public void testInvalidData() {
+                fail("Should not catch exception when checking saved image vs. original.");
+            }
+        } finally {
+            // delete the directory even if the test fails
+            if (d.exists()) {
+                deleteDirectory(OUTPUT_DIR);
+            }
+
+            ypImageSaver.close();
+        }
+    }
+    
+    @Test
+    public void testImageSavingWithLabels() throws ImageProcessingException {
+        
+        // Only run test with intended vantiq availability
+        assumeTrue(testAuthToken != null && testVantiqServer != null);
+        
+        Map config = new LinkedHashMap<>();
+        YoloProcessor ypImageSaver = new YoloProcessor();
+
+        config.put("pbFile", PB_FILE);
+        config.put("labelFile", LABEL_FILE);
+        config.put("outputDir", OUTPUT_DIR);
+        config.put("saveRate", SAVE_RATE);
+        config.put("saveImage", "local");
+        config.put("labelImage", "true");
+        try {
+            ypImageSaver.setupImageProcessing(config, MODEL_DIRECTORY, testAuthToken, testVantiqServer);
+        } catch (Exception e) {
+            fail("Could not setup the JSON YoloProcessor");
+        }
+        File d = new File(OUTPUT_DIR);
+        try {
+            // Ensure no results from previous tests
+            if (d.exists()) {
+                deleteDirectory(OUTPUT_DIR);
+            }
+
+            NeuralNetResults results = ypImageSaver.processImage(getTestImage());
+            assert results != null;
+            assert results.getResults() != null;
+            
+            // Should save first image with timestamp
+            assert d.exists();
+            assert d.isDirectory();
+            assert d.listFiles().length == 1;
+            assert d.listFiles()[0].getName().matches(timestampPattern);
+            
+            try {
+                // Converting original test image to buffered image, since this is how we save images
+                BufferedImage tempOriginal = ImageIO.read(new ByteArrayInputStream(getTestImage()));
+                ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                ImageIO.write(tempOriginal, "jpg", baos);
+                baos.flush();
+                
+                // Getting both images as byte arrays and comparing them for equality
+                byte [] originalImg = baos.toByteArray();
+                byte [] savedImg = Files.readAllBytes(d.listFiles()[0].toPath());
+                assertFalse("Images should not be identical", Arrays.equals(originalImg, savedImg));
+                baos.close();
+            } catch (IOException e) {
+                fail("Should not catch exception when checking saved image vs. original.");
+            }
+        } finally {
+            // delete the directory even if the test fails
+            if (d.exists()) {
+                deleteDirectory(OUTPUT_DIR);
+            }
+
+            ypImageSaver.close();
+        }
+    }
```

Comment:
Rather than clone all this, it would be better to have some method called (something like) ```doImageSaveTest(boolean savedLabeledImage)``` that takes a parameter to figure out whether the labels should or shouldn't be saved.  Then, the value of the diff should be the value of ```saveLabeledImage```.

Then, that just gets called two ways for the two tests.  No duplication means less to change when the time comes to change it -- which it will...